### PR TITLE
GH-2365 Add Phosphor TabPanel

### PR DESCRIPTION
**Pull request by GordonSmith**
Fixes GH-2365

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage Please Review

File: packages/phosphor/src/TabPanel.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            if (wa.widget === widget) {
+                retVal = wa;
+                return true;
+            }
+            return false;
+        });
+        return retVal;
+    }
+
+    addWidget(widget: SVGWidget | HTMLWidget, title: string) {
```
Comment (jbrundage) [inline packages/phosphor/src/TabPanel.ts]: Shouldn't this be `widget: Widget`?

**Comment by jbrundage** [comment]
Seems like there should be phosphor/src/Panel.ts that DockPanel.ts and TabPanel.ts can extend.

File: packages/phosphor/src/TabPanel.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/phosphor/src/TabPanel.ts]: Not sure why I went with that choice, I didn't test the CanvasWidget...  But will change in the future if needed.

**Pull request merged**